### ISSUE TITLE
Public release of underlying survey data & analysis

### ISSUE THREAD

**Issue by harisood** [sa-tre/satre-team#20]
### Summary of issue

At some point soon we need to release a public version of the survey data, accompanied by reproducible code with the analysis we have run on the responses.

This needs to be done carefully to ensure any public release is as safe as possible

### What needs to be done?

- [ ] Determine what of the results we can/can't publicly release
- [ ] Prepare a public facing version of the results for release
- [ ] Prepare a public facing notebook with our analytical methods

### Who can help?

* @edwardchalstrey1 
* @JimMadge 
* @manics 
* @craddm 

### Issue checklist

- [X] I have assigned the appropriate work package label to this issue, and added any other relevant labels
- [X] I have added this to the `SATRE backlog (public)` project board
- [X] I have assigned this issue to at least one person on the SATRE team

**Comment by manics** [comment]
How does this sound for the initial release?
- for each categorical question give counts of the responses
- roles: one person organises these into the high-level stakeholder groups (users, operators, builders, strategy, other), provide summary counts for each of these categories
- institutions: one person organises these into a few classes (e.g. universities, government, other public sector, private sector, whatever the person doing the work thinks is appropriate), provide summary counts for each of these categories
- other free text answers: either omit from the first version, or one person reads through and writes a summary paragraph.

In future we can consider releasing anonymised individual responses so you can look for correlations, but this is higher risk and will take longer to assess or anonymise, so should be done in a future issue.

**Comment by JimMadge** [comment]
@manics's plan sounds good to me.

The only thing I would add is to do [this](https://github.com/sa-tre/feature_survey/blob/main/example_programming_languages.ipynb) for the free text questions where there are clear categories produce counts like with the likert questions. I would hope that the function in that notebook should do most of the heavy lifting.

**Comment by harisood** [comment]
Agreed, do we want to set a date for when to have this ready by? And maybe assigning responsibility?

**Comment by manics** [comment]
I can work on:
- categorical questions
- roles
- institutions

next week

I'm happy to look at the free text answers but that'll take longer, or someone else could look at them?

**Comment by JimMadge** [comment]
I wrote the free text -> categories function, so I'm happy to do that.

I would like to have someone else pass their eyes over that though. To make sure the code is doing what we think it should, and to make sure that the choices we make (_e.g._ which responses to drop, which words to count or not count, which synonyms to use) are reasonable.

**Comment by harisood** [comment]
@manics if you have time to do free text as well that'd be huge, otherwise I can try and find some time!

**Comment by manics** [comment]
I've created a private spreadsheet for splitting roles and institutions into categories which can be made public

**Comment by harisood** [comment]
Have you had a chance to look at free text?

**Comment by manics** [comment]
I haven't

**Comment by manics** [comment]
Foillowing the last WP1 meeting the plan is to create a new spreadsheet containing the raw (row-by-row) survey results, with the following changes:
- **role** column is replaced by a role categories from the private spreadsheet
- **institution** column is replaced by an institution category from the private spreadsheet
- **categorical (Likert scale)** columns are unchanged
- **Simple free text** columns which can be automatically parsed into a list of categories (e.g. programing languages) will be replaced by those categories if possible. @JimMadge is checking- do you think we could make the single replacement column contain semi-colon separated alphabetically ordered values e.g. `java;python;r` so that this can be easily exported as CSV if we want?
- **Complex free text** columns which can't be easily summarised row-by-row will be removed, and will be summarised in future

The resulting spreadsheet will be checked for sensitive data, and if there are no problems it will be made public. The JISC online surveys JSON schema for the questionnaire will also be made available.

**Comment by JimMadge** [comment]
@manics Yes that looks possible, an example is in the [last cell here](https://github.com/sa-tre/feature_survey/blob/0f3a0c5ebeedebdcd19717988394a793361a5da3/example_programming_languages.ipynb).

**Comment by harisood** [comment]
LGTM!

**Comment by manics** [comment]
https://github.com/sa-tre/feature_survey/pull/6 is good enough
